### ISSUE TITLE
SEO and investigate making Food Oasis web-crawler-compatible

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#1356]
### Overview
Our site seems to currently require human interaction to get to listings and that our food-provider info pages aren't being crawled or indexed by search engines.

If that's the case then our site would be considered very non-SEO (search engine optimized).  

To allow people to find listings using Google and other search engines, we should consider creating a 'flat' version of our site that bots can crawl and index. 

This might be a single page (with an invisible link) with deep links to each of our organization data pages.  Those data pages should also be crawlable without requiring the react framework.

### Background

Archive.org uses web-crawlers to visit and make copies of entire sites for posterity, however it seems our site is not compatible with web-crawlers.  This means that while our landing pages are available, food-provider detail pages are not archived. https://web.archive.org/web/20220000000000*/foodoasis.la

These archives and search engine indexes are important for two reasons
- SEO: So people can find direct links to our content based on searching for an org name, for example.
- Archival purposes: So that we or others can see old versions of our data.

There is a small risk that Google may scrape our info and pull our data into their maps engine, but since our goal is to connect people with food, that's a risk we can/should take.

### Action Items
- [ ] Evaluate if our site is indexable by archive.org and google.  (Does our site work when javascript is disabled?)
- [ ] Come up with solutions that would make our site indexable, with deep links to content, so that when people search on an org name, that one of our detail pages is in the search results.  It doesn't have to be pretty.
- [ ] Implement solutions

**Comment by staceyrebekahscott** [comment]
@Nechir-89 In the future, if you would like to work on an issue that is in the New Approval column, please let one of the PMs know so we can assess and prioritize properly. I am approving this and moving to In Progress.

**Comment by staceyrebekahscott** [comment]
@Nechir-89 No worries, and there may be some workflow conflicts going on here. I'll get that sorted out with the Leads so there is no confusion.

**Comment by entrotech** [comment]
discussed on 11/14 in dev meeting. Suggested resources:
https://developers.google.com/search/docs/fundamentals/get-started-developers?hl=en
https://developers.google.com/search/docs/fundamentals/seo-starter-guide?hl=en&visit_id=638040804367265683-3650024786&rd=1
https://www.mediavine.com/mediavine-university-growing-traffic-podcast-transcript/
https://answerthepublic.com/

**Comment by Nechir-89** [comment]
@entrotech @hanapotski @staceyrebekahscott @fancyham 

I created this [figma white board](https://www.figma.com/file/jdULKEujU9tmdmXSk5tz6x/SEO?node-id=24%3A122&t=wrpZ0sVFrMplAnfe-1) for the SEO research to make work easy for all of us.

Currently within the board mainly you will see: 
- All the big steps in improving SEO (on the left side of the board).
- And how google indexed foodoasis.la (on the right side of the board).

the research is still in progress and I will add more and update to the figma board as I go,

**Comment by fancyham** [comment]
Hi everyone — I wanted to clarify that my intended main goal for this issue is to allow our org detail screens to be indexed by Google.  I’m hoping that our new deep-link feature makes this possible.

We have a ton of individual org listings — but none appear in Google’s search results.

I’m less concerned about “SEO” (at the level of keywords), though that would be a nice plus.

BTW, if anyone thinks that we shouldn’t allow our site to be indexable by Google, please post here so we can discuss.  I think it makes sense, but there are also risks as described above (for example, scrapers copying our data).

**Comment by Nechir-89** [comment]
Hey @fancyham just want to make sure I'm getting it right, from the image do you want the details panel on the left side to be indexed by google?
![screencapture-foodoasis-la-organizations-2022-11-27-23_58_09](https://user-images.githubusercontent.com/66407817/204197890-6a808767-6125-4045-90fc-580560098537.png)

**Comment by fancyham** [comment]
@Nechir-89, Yup - that's the idea.  Hopefully Google (or other search engines) can index text on the left for this org and access this org's listing via a deep-link URL.

**Comment by Nechir-89** [comment]
Thanks Bryan, I think part of the solution is to checkout if we have been indexed by google and the answer is **YES**, to prove that, right in this link you can see almost all pages that have been indexed by google for foodoasis.la 

[food oasis's index pages by google](https://www.google.com/search?q=site:foodoasis.la/&ei=XxV5Y8bLA_3ZkPIP3q--oAM&start=0&sa=N&ved=2ahUKEwiG9-ay5br7AhX9LEQIHd6XDzQ4ChDy0wN6BAgCEAQ&biw=1858&bih=977&dpr=1) 

the problem as you mentioned earlier we can not see any pantry, food provider, or even food program in any of these indexed pages by google, I'm just doing research to find solution for this problem

**Comment by fancyham** [comment]
I suspect we'd need to do something like 'server-side rendering' to flatten out the website for the Google Indexer?  Sounds like Google's indexer doesn't work well with ReactJS, so this basically creates dummy deep-link pages that it can index.

https://stackoverflow.com/a/52039854/3366821

**Comment by fancyham** [comment]
Hi @Nechir-89, Just checking in to see if you've discovered anything that can help or run into roadblocks?  (I just heard some Google Analytics results which reminded me about this and the free marketing that Google Search could give us, but currently we're missing out on)

**Comment by jelenaUX** [comment]
Wearing my UX hat: It seems to me that Google search indexing is affecting the way that FOLA visitors behave. ("Visitors" = US-based people who visit FOLA website on their phone and do not click on any obvious FOLA staff/volunteer links.) Based on Google Analytics data, most visitors now arrive by directly typing in foodoasis.la into their URL bar and then proceed to look for pantries. The only informational page that visitors consult is "About Us" page, perhaps because it is prominently featured in Google search results for FOLA. Only a small number of visitors (10%) click on "About Us" and for the most part these visitors do not continue to look for pantry info. So the following changes may increase the numbers of visitors to the website and improve their experiences: 
- Making it possible for Google to index specific food pantries on FOLA website;
- Improving SEO description of "About Us" page;
- Adding a button "Find Food" into the main text on the "About Us" page.

**Comment by jelenaUX** [comment]
Here is a brief (3-page) [cohort report](https://docs.google.com/presentation/d/1Z5RM0TEiCqP1aQyKfXKjy1GPJF3kfyrAYz3ItOhewTs/edit#slide=id.g22bd4372a04_0_12) with my interpretation. It looks like SEO would likely result in many more organic visitors, and these visitors have great conversion rates.

**Comment by Nechir-89** [comment]
Hi everyone,

For this issue I already discussed with @entrotech @hanapotski , and I believe this issue needs to be breaking down into smaller tasks and issues, Google already indexed us very well for about 15 pages and we have older versions of FOA in Archive.org

**Comment by fancyham** [comment]
Okay, just a reminder that this issue’s goal is to get the individual listings indexed not the static pages (which are already indexed fine).  

Use case:
- Someone googles for “food pantries in west covina” — it’d be great to have a FOLA list of neighborhood pantries appear in Google’s search results.

**Comment by fancyham** [comment]
Hi @Nechir-89 - would it make sense to un-assign yourself from this issue at this point so it moves to the correct kanban column?

**Comment by Nechir-89** [comment]
Okay, I'll unassign myself from it!

**Comment by fancyham** [comment]
Hi there - checking in on this older but important issue — what's the status on this currently?

**Comment by fancyham** [comment]
[This may be as simple as replacing our JavaScript buttons with explicit ahrefs.](https://support.google.com/webmasters/thread/15430775/hi-is-seo-going-to-be-an-issue-with-a-website-made-on-react-website-has-multiple-pages?hl=en)

**Comment by fancyham** [comment]
Not sure why bot closed this

**Comment by Giolii** [comment]
Hey everyone, I talked with an SEO expert at my company about this.

From what she said, it makes sense that the homepage is doing better on Google than the individual organization pages. The homepage has the stronger keywords and probably gets most of the backlinks, while the individual listings may not have enough search volume by themselves.

One idea she suggested is to add a crawler-friendly link from the homepage to a page that shows the listings in a way Google can index. Something like “All Organizations” or “Neighborhoods.”

I think neighborhoods might be a good place to start, since that’s probably closer to how someone would search, like “food pantries in Venice” or “free food in West Covina.” We could start with a simple version, check if Google indexes it with Search Console, and then improve the link structure from there.

If it works, we can later add more neighborhood pages, or maybe even direct links to individual listings.

Some of my ideas: 

<img width="1038" height="535" alt="Image" src="https://github.com/user-attachments/assets/8ca3feef-ecde-4b9c-8b13-53ea8049a31b" />
<img width="627" height="535" alt="Image" src="https://github.com/user-attachments/assets/f7f6d5f5-f164-41e6-8fab-08a878397732" />
